daemon/endpoint.go

```diff
@@ -582,9 +582,15 @@ func (d *Daemon) deleteEndpointQuiet(ep *endpoint.Endpoint, releaseIP bool) []er
 	}
 	ep.SetStateLocked(endpoint.StateDisconnecting, "Deleting endpoint")
 
+	// Since the endpoint is being deleted, we no longer need to run events
+	// in its event queue.
+	ep.CloseEventQueue()
+
+	wg := d.policy.RemoveIdentifierFromRuleCaches(ep.ID)
+
 	// Remove the endpoint before we clean up. This ensures it is no longer
 	// listed or queued for rebuilds.
-	endpointmanager.Remove(ep)
+	endpointmanager.Remove(ep, wg)
```

Comment:
Note that we need to wait that identifier is removed from the caches within rules?